lib/msal-common/src/response/ResponseHandler.ts

```diff
@@ -151,16 +152,47 @@ export class ResponseHandler {
     /**
      * Function which validates server authorization token response.
      * @param serverResponse
+     * @param refreshAccessToken
      */
     validateTokenResponse(
-        serverResponse: ServerAuthorizationTokenResponse
+        serverResponse: ServerAuthorizationTokenResponse,
+        refreshAccessToken?: boolean
     ): void {
         // Check for error
         if (
             serverResponse.error ||
             serverResponse.error_description ||
             serverResponse.suberror
         ) {
+
+            // check if 500 error
+            if (
+                refreshAccessToken &&
+                serverResponse.status &&
+                (serverResponse.status >= ErrorCodeBounds.stsErrorLowerLimit) &&
+                (serverResponse.status <= ErrorCodeBounds.stsErrorUpperLimit)
+            ) {
+                this.logger.warning(
```

Comment:
Make sure to include as much information as possible about the error in the log message.